### PR TITLE
YARA-1550: Yara42 new math attributes

### PR DESCRIPTION
**Pull request by TadeasKucera**
Mirrors the changes in yara introduced here:
https://github.com/VirusTotal/yara/commit/87bc03d58743aaaa0f11b6d19d1d26e717f3cb01
https://github.com/VirusTotal/yara/commit/0d5f2cd5ef2595d93886db05d970107f83c494d7
https://github.com/VirusTotal/yara/commit/2edd10c735d91120f0f567c893093c873a67ba95
Solves YARA-1550, YARA-1552 and YARA-1556.

### CODE REVIEW

File: tests/cpp/parser_tests.cpp [hunk 1]
```diff
@@ -3521,17 +3521,17 @@ import "math"
 rule math_module
 {
 	condition:
-		math.entropy("dummy") > 7
+		math.to_number(math.entropy("dummy") > 7) == 1
```
Comment (metthal) [inline tests/cpp/parser_tests.cpp:3524]: Should this be a brand new test instead of rewriting existing one?

Comment (TadeasKucera) [reply tests/cpp/parser_tests.cpp:3524]: We do not have a test for each module attribute, so I personally think there is no trouble with testing more functions within one test. But I can create a new test if you would prefer it.

Comment (metthal) [reply tests/cpp/parser_tests.cpp:3524]: Yeah, just put there more attributes.

Comment (TadeasKucera) [reply tests/cpp/parser_tests.cpp]: Done :+1:

**Review by metthal — APPROVED** [review]

**Pull request merged**